src/components/buttonSelectors/NumberButtonSelector.js

```diff
@@ -1,90 +1,171 @@
+        case 2:
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: val,
+              vehicleID: vehicleID,
+            });
+          break;
+        case 3:
+          // Case 3 updates the passenger form
+          let passengerID = dependencyID[2];
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: val,
+              vehicleID: vehicleID,
+              passengerID: passengerID,
+            });
+        case 4:
+          // Case 4 updates the nonmotorist form
+          updateResponse &&
+            updateResponse({
+              id,
+              question: currId,
+              selection: val,
+              vehicleID: vehicleID,
+              nonmotoristID: dependencyID[3],
+            });
+        default:
+          break;
+      }
     }
+    // Saves the selected answer to the JSON file
+    submitFunction({ id, question: currId, selection: val });
+  };
 
-    const tooltip = () => {
-        return(<TooltipView toolTip={data.tooltip} helperImg={data.helperImg}/>)
-    }
+  // Renders tooltip component
+  const tooltip = () => {
+    return <TooltipView toolTip={data.tooltip} helperImg={data.helperImg} />;
+  };
 
-    return (
-        <QuestionSection title={data.question} helperText={data.helperText} tooltip={tooltip()} required={data.required}>
-            <ScrollView horizontal={true} showsHorizontalScrollIndicator={true} contentContainerStyle={styles.container}>
-                <HStack space={4}>
-                    <Button variant={selection === '0' ? 'solid': 'subtle'} onPress={() => submitField('0')}>None</Button>
-                    <Button variant={selection === '1' ? 'solid': 'subtle'} onPress={() => submitField('1')}>1</Button>
-                    <Button variant={selection === '2' ? 'solid': 'subtle'} onPress={() => submitField('2')}>2</Button>
-                    <Button variant={selection === '3' ? 'solid': 'subtle'} onPress={() => submitField('3')}>3</Button>
-                    <Button variant={selection === '4' ? 'solid': 'subtle'} onPress={() => submitField('4')}>4</Button>
-                    <Button variant={selection === '5' ? 'solid': 'subtle'} onPress={() => submitField('5')}>5</Button>
-                    <Input
-                        placeholder = "Other"
-                        onChangeText = {submitField}
-                        value = {(selection > 5) ? selection:''}
-                        size="md"
-                        ml={4}
-                    />
-                </HStack>
-            </ScrollView>
-        </QuestionSection>
-    )
-}
+  let renderComponent = true;
+  // Check if the question has any dependencies to update
+  if (data.questionDependency != undefined && props.response != null) {
+    renderComponent = dependencyParser(props.response, data, dependencyID);
+  }
+
```

Comment:
wondering if we want to add an if statement that if renderComponent then return the button selector otherwise return an empty <>? right now it doesn't seem like renderComponent is being used at all, which is not its intended purpose